### PR TITLE
examples and artists cleanup

### LINKED ISSUE #397: plotting with global colorbar seems to be broken in wt.artists.mpl_2D.
```
import WrightTools as wt

p = '000.data'
d = wt.data.from_PyCMDS(p)

d.signal_ratio.signed = True

art = wt.artists.mpl_2D(d, xaxis='w2', yaxis='w1')
art.plot(channel='signal_ratio', local=False, pixelated=True)
```
[000.data.zip](https://github.com/wright-group/WrightTools/files/1536595/000.data.zip)

Additionally, `art.plot(channel='signal_ratio', pixelated=False)` plots figures with a signed colorbar but default colormap in the plot.

### LINKED ISSUE #431: overloaded pcolor should call pcolor_helper when handed bare arrays
We currently overload `pcolor` and allow the method to be handed a data object or bare arrays. When handed bare arrays, the method does not call `pcolor_helper`. Do we desire it to call `pcolor_helper`? 

We could even be sly and do some dimensionality checks behind the scene to see if the user has already accomplished the regridding. For instance, if the `x`, `y`, `z` arrays have shapes `(M, N), (M, N), (M-1, N-1)` then `pcolor_helper` is not called within our overloaded `pcolor` method, but if these shape requirements are not met, `pcolor_helper` is called.

### LINKED ISSUE #485: fix broken examples

### LINKED ISSUE #486: clean-up artists
mark broken artists as notimplemented
delete code that has no future

### LINKED ISSUE #496: pcolor_helper in-place changes the shape of 1D input arrays
`wt.artists.pcolor_helper` in-place changes the shape of 1D input arrays. Is this desired/expected behavior? I was not expecting this behavior. It changes the shape of the arrays s.t. it isn't a problem for some uses, but it is if you want to index into the arrays after calling `pcolor_helper` and you are expecting 1D arrays. 

Example script:
```
import numpy as np
import WrightTools as wt

x = np.arange(0, 10, 1)
y = np.arange(0, 10, 1)
z = x[:, None] * y[None, :]
print(x.shape, y.shape, z.shape)
X, Y, Z = wt.artists.pcolor_helper(x, y, z)
print(x.shape, y.shape, z.shape)
print(X.shape, Y.shape, Z.shape)
```
Results in:
```
(10,) (10,) (10, 10)
(10, 1) (1, 10) (10, 10)
(11, 11) (11, 11) (10, 10)
```

### PR DESCRIPTION
**Pull request by untzag**
resolves #485
resolves #496
resolves #486
resolves #397 
resolves #431

### CODE REVIEW

File: WrightTools/artists/_base.py [hunk 1]
```diff
@@ -45,6 +45,9 @@ class Axes(matplotlib.axes.Axes):
     transposed = False
     is_sideplot = False
 
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
```
Comment (ksunden) [inline WrightTools/artists/_base.py]: Why have an explicit `__init__` if it just calls super?

File: WrightTools/data/_data.py [hunk 2]
```diff
@@ -327,10 +327,11 @@ def chop(self, *args, at={}, parent=None, verbose=True):
         args = list(args)
         for i, arg in enumerate(args):
             if isinstance(arg, int):
-                args[i] = self._axes[i].expression
+                args[i] = self._axes[arg].expression
         # get output collection
         out = wt_collection.Collection(name='chop', parent=parent)
         # get output shape
+        print(args)
```
Comment (ksunden) [inline WrightTools/data/_data.py]: Print

File: WrightTools/data/_data.py [hunk 3]
```diff
@@ -670,11 +670,15 @@ def level(self, channel, axis, npts, *, verbose=True):
...
+        print(npts)
         if npts > 0:
             ss[axis] = slice(0, npts, None)
         else:
-            ss[axis] = slice(-npts, None, None)
+            ss[axis] = slice(npts, None, None)
+        print(ss)
+        print(channel[ss].shape)
         subtrahend = np.nanmean(channel[ss], axis=axis)
+        print(subtrahend.shape)
```
Comment (ksunden) [inline WrightTools/data/_data.py]: Prints

**Comment by darienmorrow** [comment]
Is additional documentation needed for the overloaded `pcolor`? The new version seems to smartly use `pcolor_helper` so that it doesn't need to be called outside of the method. Should the docstring reflect this capability?

**Comment by untzag** [comment]
@darienmorrow thanks for the suggestion---done

File: WrightTools/artists/_helpers.py [hunk 4]
```diff
@@ -0,0 +1,917 @@
...
+        for idx in np.ndindex(out.shape):
+            ul = orig[idx[0] + 1, idx[1] + 0]
+            ur = orig[idx[0] + 1, idx[1] + 1]
+            ll = orig[idx[0] + 0, idx[1] + 0]
+            lr = orig[idx[0] + 0, idx[1] + 1]
+            out[idx] = np.mean([ul, ur, ll, lr])
+    return X, Y, zi
+
+
+def plot_colorbar(cax=None, cmap='default', ticks=None, clim=None, vlim=None,
```
Comment (ksunden) [inline WrightTools/artists/_helpers.py]: should this be in `_helpers` or `_colors`?

File: examples/fringes_transform.py [hunk 5]
```diff
@@ -14,20 +14,24 @@
 p = datasets.PyCMDS.w2_w1_000
 data = wt.data.from_PyCMDS(p)
 
+data.signal_mean **= 0.5  # to amplitude level
```
Comment (ksunden) [inline examples/fringes_transform.py]: should we use the kit method?

File: WrightTools/artists/_base.py [hunk 6]
```diff
@@ -750,838 +557,4 @@ def apply_rcparams(kind='fast'):
...
-                writer.append_data(image)
-    except BaseException:
-        print('Error: {0}'.format(sys.exc_info()[0]))
-        return None
-    # finish
-    if verbose:
-        interval = np.round(t.interval, 2)
-        print('gif generated in {0} seconds - saved at {1}'.format(interval, outpath))
-    return outpath
+from ._helpers import pcolor_helper  # flake8: noqa (circular import)
```
[the remark this answers is not in the document]
Comment (untzag) [reply WrightTools/artists/_base.py]: circular import :cry:

File: WrightTools/artists/_helpers.py (hunk 4 above)
Comment (untzag) [reply WrightTools/artists/_helpers.py]: I think I do like it in _helpers

File: examples/fringes_transform.py (hunk 5 above)
Comment (untzag) [reply examples/fringes_transform.py]: yes! thank you

File: WrightTools/artists/_colors.py [hunk 7]
```diff
@@ -11,11 +11,13 @@
...
 import matplotlib.pyplot as plt
 import matplotlib.colors as mplcolors
+import matplotlib.gridspec as grd
+
 
 # --- define -------------------------------------------------------------------------------------
 
 
-__all__ = ['colormaps']
+__all__ = ['colormaps', 'get_color_cycle', 'grayify_cmap', 'plot_colormap_components']
```
[the remark this answers is not in the document]
Comment (untzag) [reply WrightTools/artists/_colors.py]: thanks

**Pull request merged**